crates/sui-core/src/authority.rs

```diff
@@ -1111,6 +1112,34 @@ impl AuthorityState {
         SuiTransactionEffects::try_from(effects, self.module_cache.as_ref())
     }
 
+    pub async fn dev_inspect_transaction(
+        &self,
+        transaction: TransactionData,
```

Comment:
`TransactionData` requires gas payments info like gas payment object and budget, which seems not necessary for dryRunMoveCall. iiuc we will expose 2 endpoints:
- dryRunTxn, for which `dev_inspect_transaction` is perfect
- dryRunMoveCall, where it would be great to have a function like this for a single move call without txn wrapper
```
fn dev_inspect_move_call(&self, move_call: MoveCall) -> Result<ExecutionResult>
```
@sblackshear any wisdom here?

Reply:
I think that seems reasonable here. Maybe we can even land this PR and iterate on the way the endpoints are exposed?

FWIW, I'm trying to avoid the term "dry run" here, hence the new term "dev inspect". I think we should still have dry run, but it should work literally exactly the same as the normal execution (except for committing effects). So that end point would mostly be used for sanity checking and gas estimation. 
Whereas "dev inspect" would be a much more powerful tool that could be used for all sorts of things, but the developer needs to be aware that they might not actually be able to run a similar transaction.

Reply:
Agreed that this sounds reasonable!

Reply:
Re if we want both dryRunTxn and devInspectTxn, I think devInspectTxn is a better impl. and thus I would prefer to have one public RPC endpoint instead of 2.

I am neutral about the naming, either keeping dryRun or use devInspect sounds good to me, as long as this prefix is applied to both txn execution and move call.

Reply:
The 3 types of transactions I see:
- Normal. Used for making state changes to the system 
- Dry Run. Exactly the same as Normal, but does not make state changes. Useful for testing gas or testing the changes a transaction will make
- Dev Inspect. The wild west of RPC endpoints. Will let you call any Move function with just any arbitrary values. 

The reason I think we should have Dry Run in addition is that not all "Dev Inspect" transactions can be run as Normal transactions. So you might end up thinking your txn will cost X gas, and you go and submit it and it gets rejected with some small gas payment. In other words, I think for the best developer experience, we need some endpoint that behaves just like a "Normal" transaction, but without committing changes.